mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -442,14 +440,52 @@ struct BlockwiseGemmV2RewritePattern
         sourceOffset = kb.create<MulIOp>(
             loc, sourceOffset, kb.create<ConstantIndexOp>(loc, KPack));
 
-      Value destOffset = k_i;
-
-      // Emit load/store
+      Type loadType = vectorTypeOrSelf(dataType, KPack);
+      Value value =
+          kb.create<InBoundsLoadOp>(loc, loadType, ldsOrig, sourceOffset);
       auto bufferType = regDest.getType().cast<MemRefType>();
       Type bufferElementType = bufferType.getElementType();
-      Value value = kb.create<InBoundsLoadOp>(loc, bufferElementType, ldsOrig,
-                                              sourceOffset);
-      kb.create<memref::StoreOp>(loc, value, regDest, ValueRange{destOffset});
+
+      // We're loading in units of kPack, but storing in units of k_base.
```

Comment:
So you are effectively vectorizing the load/store in lds -> register transfer? That's good motivation!

Reply:
We already had this vectorized, I think

Reply:
... also, the alluded-to "bug" was actually the gpu on my favorite Lockhart being down

Reply:
So you are saying although previously we only load and store 1 element, the backend will consolidate the load/store a vector?

Reply:
The comment above about the "bug" was in reference to a previous commit - I put it in the wrong place.

Now, as to this comment, blockwise gemm already loaded vectors of length kPack from LDS.

Reply:
Ah right the unit used to be kPack.